### PR TITLE
Add support for batch size 0

### PR DESCRIPTION
**Pull request by dreiss**

### CODE REVIEW

File: src/operator-run.c [hunk 1]
```diff
@@ -606,6 +606,11 @@ static void compute_u8softargmax(
 
 enum qnnp_status qnnp_run_operator(qnnp_operator_t op, pthreadpool_t threadpool)
 {
+  // For any ukernel type, there is no work to do if the batch size is 0.
+  if (op->batch_size == 0) {
+    return qnnp_status_success;
```
Comment (supriyar) [inline src/operator-run.c]: In what case will we receive batch size 0? Should we be emitting an info message here saying we got batch size 0, or is that done in the C2 land?

File: test/channel-shuffle.cc [hunk 2]
```diff
@@ -11,6 +11,15 @@
 #include "channel-shuffle-operator-tester.h"
 
 
+TEST(CHANNEL_SHUFFLE_OP, zero_batch) {
+  ChannelShuffleOperatorTester()
+    .batchSize(1)
```
Comment (supriyar) [inline test/channel-shuffle.cc]: Did you mean batchSize(0) here?

File: test/convolution-operator-tester.h [hunk 3]
```diff
@@ -402,8 +401,11 @@ class ConvolutionOperatorTester {
           }
         }
       }
-      const int32_t accumulatorsMin = *std::min_element(accumulators.cbegin(), accumulators.cend());
-      const int32_t accumulatorsMax = *std::max_element(accumulators.cbegin(), accumulators.cend());
+      // Create dummy min/max for empty inputs.
+      // These are only used to compute scale and zero point,
+      // and real callers will just pull those values from the model.
+      const int32_t accumulatorsMin = accumulators.empty() ?   0 : *std::min_element(accumulators.cbegin(), accumulators.cend());
+      const int32_t accumulatorsMax = accumulators.empty() ? 900 : *std::max_element(accumulators.cbegin(), accumulators.cend());
```
Comment (supriyar) [inline test/convolution-operator-tester.h]: Why 900? I guess it doesn't really matter but just curious.

File: src/operator-run.c (hunk 1 above)
Comment (dreiss) [reply src/operator-run.c]: I don't think we should be emitting a message in any case, just like Python doesn't emit a message if you try to do a comprehension over an empty list.

File: test/channel-shuffle.cc (hunk 2 above)
Comment (dreiss) [reply test/channel-shuffle.cc]: Yep.  Oops.

File: test/convolution-operator-tester.h (hunk 3 above)
Comment (dreiss) [reply test/convolution-operator-tester.h]: I don't actually remember.  I think I tried 1 or 255 initially and had some problems.

File: src/convolution.c [hunk 4]
```diff
@@ -177,7 +177,7 @@ enum qnnp_status qnnp_create_convolution2d_nhwc_q8(
 
   const size_t kernel_size = kernel_height * kernel_width;
 
-  enum qnnp_ukernel_type ukernel_type = qnnp_ukernel_type_none;
+  enum qnnp_ukernel_type ukernel_type;
```
[the remark this answers is not in the document]
Comment (dreiss) [reply src/convolution.c]: Okay.  I'll pull this commit out.  I originally was changing the semantics of `qnnp_ukernel_type_none`, so I wanted to eliminate a spurious usage of it.

**Comment by dreiss** [comment]
> Does batch size of 0 mean processing reduces to an empty loop effectively?

In principle, yes, but in this diff I implemented it as an early bail out because I had some difficulty getting some of the other parameters computed properly for batch zero.  When we make the interface more functional, I hope we won't need these special cases.

**Pull request merged**